custodia/ipa/vault.py

```diff
@@ -71,26 +70,54 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 
 class IPAVault(CSStore):
-    principal = PluginOption(str, 'custodia/' + FQDN, "Principal for auth")
+    # vault arguments
+    principal = PluginOption(str, None, "Service principal for service vault")
+    user = PluginOption(str, None, "User name for user vault")
+    vault_type = PluginOption(
+        str, "service", "vault type, one of 'user', 'service', or 'shared'")
+
+    # Kerberos flags
     krb5config = PluginOption(str, None, "Kerberos krb5.conf override")
     keytab = PluginOption(str, None, "Kerberos keytab for auth")
     ccache = PluginOption(
         str, None, "Kerberos ccache, e,g. FILE:/path/to/ccache")
+
+    # ipalib.api arguments
     ipa_confdir = PluginOption(str, None, "IPA confdir override")
     ipa_context = PluginOption(str, "cli", "IPA bootstrap context")
     ipa_debug = PluginOption(bool, False, "debug mode for ipalib")
 
     def __init__(self, config, section=None):
         super(IPAVault, self).__init__(config, section)
+
+        if self.vault_type == 'shared':
+            self._vault_args = {'shared': True}
+        elif self.vault_type == 'user':
+            if self.user is None:
+                msg = "{!r}: User vault requires 'user' parameter"
```

Comment:
No idea about delegation, will we have information about it in IPA? Or will it be ad hoc configuration on the client side ?